docs/sources/logql/functions.md

```diff
@@ -0,0 +1,370 @@
+
+```template
+{{ .path }}
+```
+
+You can take advantage of [pipeline](https://golang.org/pkg/text/template/#hdr-Pipelines) to chain multiple functions.
+In a chained pipeline, the result of each command is passed as the last argument of the following command.
+
+Example:
+
+```template
+{{ .path | replace " " "_" | trunc 5 | upper }}
+```
+
+## ToLower & ToUpper
+
+Convert the entire string to lowercase or uppercase.
+
+Signatures:
+
+- `ToLower(string) string`
+- `ToUpper(string) string`
+
+Examples:
+
+```template
+"{{.request_method | ToLower}}"
+"{{.request_method | ToUpper}}"
+`{{ToUpper "This is a string" | ToLower}}`
+```
+
+> In Loki 2.1 you can also use respectively [`lower`](#lower) and [`upper`](#upper) shortcut, e.g `{{.request_method | lower }}`
+
+## Replace string
+
+> in Loki 2.1 [`replace`](#replace) (as opposed to `Replace`) is available with a different signature but easier to chain within pipeline.
+
+Perform simple string replacement.
+
+Signature:
+
+`Replace(s, old, new string, n int) string`
+
+It takes four arguments:
+
+- `s` source string
+- `old` string to replace
+- `new` string to replace with
+- `n` the maximun amount of replacement (-1 for all)
+
+Example:
+
+```template
+`{{ Replace "This is a string" " " "-" -1 }}`
+```
+
+The above will produce `This-is-a-string`
+
+## Trim, TrimLeft, TrimRight and TrimSpace
+
+> In Loki 2.1 [trim](#trim), [trimAll](#trimAll), [trimSuffix](#trimSuffix) and [trimPrefix](trimPrefix) have been added with a different signature for better pipeline chaining.
+
+`Trim` returns a slice of the string s with all leading and
+trailing Unicode code points contained in cutset removed.
+
+Signature: `Trim(value, cutset string) string`
+
+`TrimLeft` and `TrimRight` are the same as `Trim` except that it respectively trim only leading and trailing characters.
+
+```template
+`{{ Trim .query ",. " }}`
+`{{ TrimLeft .uri ":" }}`
+`{{ TrimRight .path "/" }}`
+```
+
+`TrimSpace` TrimSpace returns string s with all leading
+and trailing white space removed, as defined by Unicode.
+
+Signature: `TrimSpace(value string) string`
+
+```template
+{{ TrimSpace .latency }}
+```
+
+`TrimPrefix` and `TrimSuffix` will trim respectively the prefix or suffix supplied.
+
+Signature:
+
+- `TrimPrefix(value string, prefix string) string`
+- `TrimSuffix(value string, suffix string) string`
+
+```template
+{{ TrimPrefix .path "/" }}
+```
+
+## regexReplaceAll & regexReplaceAllLiteral
+
+`regexReplaceAll` returns a copy of the input string, replacing matches of the Regexp with the replacement string replacement. Inside string replacement, $ signs are interpreted as in Expand, so for instance $1 represents the text of the first sub-match. See the golang [docs](https://golang.org/pkg/regexp/#Regexp.ReplaceAll) for detailed examples.
```

Comment:
Make the link text a more descriptive word than "docs". It would ideally be the title of the link target.